TombEngine/Specific/BitField.cpp

```diff
@@ -0,0 +1,286 @@
+#include "framework.h"
+#include "Specific/BitField.h"
+
+#include <limits>
+
+using std::string;
+using std::vector;
+
+namespace TEN::Utils
+{
+	constexpr auto BIT_FIELD_SIZE_MAX = std::numeric_limits<uint>::digits;
+
+	BitField::BitField()
+	{
+		this->Bits.resize(BIT_FIELD_SIZE_MAX);
+	}
+
+	BitField::BitField(uint size)
+	{
+		size = std::min<uint>(size, BIT_FIELD_SIZE_MAX);
+		this->Bits.resize(size);
+	}
+
+	BitField::BitField(uint size, uint packedBits)
+	{
+		size = std::min<uint>(size, BIT_FIELD_SIZE_MAX);
+		this->Bits.resize(size);
+
+		for (uint i = 0; i < size; i++)
+		{
+			uint bit = uint(1 << i);
+			if ((packedBits & bit) == bit)
+				this->Bits[i] = true;
+		}
+	}
+
+	BitField::BitField(const string& bitString)
+	{
+		for (const char& bit : bitString)
+			this->Bits.push_back((&bit == "1") ? true : false); // TODO: Check.
+	}
+
+	uint BitField::GetSize()
+	{
+		return Bits.size();
+	}
+
+	uint BitField::GetCount()
+	{
+		uint count = 0;
+		for (const uint& bit : this->Bits)
```

Comment:
this->Bits is vector of bool, so it should be:

`for (const bool& bit : this->Bits)`